project/src/components/Form.js

```diff
@@ -0,0 +1,251 @@
+import React, {Component} from 'react'
+import { connect } from 'react-redux'
+import { addValue } from '../actions/index'
+import {withRouter} from 'react-router-dom'
+import $ from 'jquery'
+
+import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import '../css/style.css';
+const Validation = require ('../components/validationComponent.js');
+const {addUserUrl} = require('../constants');
+
+const myRouterComponent = withRouter (class Form extends Component {
+
+    render(){
+            let nameField;
+            let surNameField;
+            let emailField;
+            let ageField;
+            let middleNameField;
+        let imgFile;
+
+        let onNameChange= (e)=>{
+                let val = e.target.value;
+                let valid = Validation.validateName(nameField.value);
+                this.props.dispatch(addValue('name', val, valid));
+            if(valid){
```

Comment:
``` if(valid){
                 e.target.classList.remove('is-invalid');
                    e.target.classList.add('is-valid');
                } else{
                    e.target.classList.add('is-invalid');
                    e.target.classList.remove('is-valid');
                }
```
I saw this part of code few times in your code. You definitely have to move it to the separate function and invoke that function here.